app/util/data_preparation/bitbucket/prepare-data.py

```diff
@@ -19,15 +19,31 @@ def generate_random_string(length=20):
     return "".join([random.choice(string.ascii_lowercase) for _ in range(length)])
 
 
-def __get_users(bitbucket_api):
-    perf_user_count = BITBUCKET_SETTINGS.concurrency
-    perf_users = bitbucket_api.get_users(username=f'{DEFAULT_USER_PREFIX}', max_results=perf_user_count)
-    perf_user_count_to_create = perf_user_count - len(perf_users)
+def __get_admin_users(client: BitbucketRestClient):
+    perf_users = __get_users(client)
+    for user in perf_users:
+        client.change_user_permissions(user['name'], BitbucketUserPermission.ADMIN)
+
+    return perf_users
+
+
+def __get_users(client: BitbucketRestClient):
+    perf_users_desired_number = BITBUCKET_SETTINGS.concurrency
+    current_perf_users = client.get_users(f'{DEFAULT_USER_PREFIX}', perf_users_desired_number)
+    perf_users_current_number = len(current_perf_users)
+    if not (perf_users_current_number < perf_users_desired_number):
+        return current_perf_users
+
+    perf_user_count_to_create = perf_users_desired_number - perf_users_current_number
     while perf_user_count_to_create > 0:
-        user = bitbucket_api.create_user(username=f'{DEFAULT_USER_PREFIX}-{generate_random_string(5)}')
-        perf_users.append(user)
+        client.create_user(f'{DEFAULT_USER_PREFIX}-{generate_random_string(5)}')
         perf_user_count_to_create = perf_user_count_to_create - 1
-    return bitbucket_api.get_users(username=f'{DEFAULT_USER_PREFIX}', max_results=perf_user_count)
+
+    perf_users = client.get_users(f'{DEFAULT_USER_PREFIX}', perf_users_desired_number)
+    if len(perf_users) < perf_users_desired_number:
+        raise SystemExit(f'Server returned less number of users than expected')
```

Comment:
we could add variables into log message

Reply:
it is a very rare case and we could have the numbers from users in csv file and concurrency in yml file